src/Frontend/util/use-debounced-input.ts

```diff
@@ -0,0 +1,23 @@
+// SPDX-FileCopyrightText: Meta Platforms, Inc. and its affiliates
+// SPDX-FileCopyrightText: TNG Technology Consulting GmbH <https://www.tngtech.com>
+//
+// SPDX-License-Identifier: Apache-2.0
+import { useEffect, useState } from 'react';
+
+const default_delay = 500;
```

Comment:
global constants in typescript are usually all uppercase:
```suggestion
const DEFAULT_DELAY = 500;
```